misc/completion/ipfs-completion.bash

```diff
@@ -1,421 +1,84 @@
-_ipfs_comp()
-{
-    COMPREPLY=( $(compgen -W "$1" -- ${word}) )
-}
-
-_ipfs_help_only()
-{
-    _ipfs_comp "--help"
-}
-
-_ipfs_add()
-{
-    _ipfs_comp "--recursive --quiet --progress --trickle --only-hash
-            --wrap-with-directory --hidden --help"
-}
-
-_ipfs_bitswap()
-{
-    ipfs_comp "stat wantlist --help"
-}
-
-_ipfs_bitswap_stat()
-{
-    _ipfs_help_only
-}
-
-_ipfs_bitswap_wantlist()
-{
-    ipfs_comp "--peer= --help"
-}
-
-_ipfs_block()
-{
-    _ipfs_comp "get put stat --help"
-}
-
-_ipfs_block_get()
-{
-    _ipfs_help_only
-}
-
-_ipfs_block_put()
-{
-    _ipfs_help_only
-}
-
-_ipfs_block_stat()
-{
-    _ipfs_help_only
-}
+# License: MIT
 
-_ipfs_bootstrap()
+_ipfs_comp ()
 {
-    _ipfs_comp "add list rm --help"
-}
+	local reply
+	case "$cur" in
+		-*)	reply="$(__ipfs_flag_list $*)
+				--help" ;;
+		*)	reply="$(__ipfs_sub_list $*)" ;;
+	esac
 
-_ipfs_bootstrap_add()
-{
-    _ipfs_comp "--default --help"
-}
-
-_ipfs_bootstrap_list()
-{
-    _ipfs_help_only
-}
+	# no more sub-commands
+	if [[ -z "$reply" ]]; then
+		local last=${cseq[-1]}
+		case "$last" in
+			id|ping)
+				reply="$(__ipfs_peers_list)"
+				compopt -o nospace ;;
+			cat|ls)
+			# TODO: flesh this case out to handle completing within
+			# a directory node.
+				reply="$(__ipfs_pin_list)"
+				compopt -o nospace ;;
+			add|replace)
+				compopt -o default ;;
+		esac
+	fi
 
-_ipfs_bootstrap_rm()
-{
-    _ipfs_comp "--all --help"
+	COMPREPLY=( $(compgen -W "$reply" -- "$cur") )
+	return 0
 }
 
-_ipfs_cat()
-{
-    _ipfs_help_only
+__ipfs_pin_list() {
+	ipfs refs local 2>/dev/null
```

Comment:
I would be careful about using this, its not a fast call (especially when you have gigabytes of content)